### ISSUE TITLE
Does Macchina support two batteries?

### ISSUE THREAD

**Issue by mohabmetwalli** [Macchina-CLI/macchina#295]
I wasn't sure if I should open this issue as a bug or a feature request, so I thought I should ask this question first.

My laptop has two batteries, and I suspect this is the reason `macchina` cannot read battery percentage.

`macchina --doctor` output:

```
Let's check your system for errors... Here's a summary:

We've collected 9 readouts, including 1 failed read(s) and 0 read(s) which resulted in a warning.
Readout "Battery" failed with message: No such file or directory (os error 2)
```

`ls /sys/class/power_supply --all` output:

```
.  ..  AC  BAT0  BAT1  ucsi-source-psy-USBC000:001  ucsi-source-psy-USBC000:002
```

**Comment by FantasyTeddy** [comment]
Even if multiple batteries are present, shouldn't macchina just report the status of the first one?

@mohabmetwalli: Can you show the directory listing of `/sys/class/power_supply/BAT0`?

**Comment by grtcdr** [comment]
> Even if multiple batteries are present, shouldn't macchina just report the status of the first one?

Yes, indeed. I'll see what I can do to fix this. Adding support for this could prove to be a lot of work in `libmacchina` — dual-battery friends, if you really want to see support for multi-battery reports, please contribute your patches to `libmacchina` :)

**Comment by naguam** [comment]
Hi,

Regarding batteries, this library might be helpful,

https://crates.io/crates/starship-battery

It has one API which lists and works with most major operating systems (including NetBSD thanks to my port) except OpenBSD (For now, I wanted to do it but the rust ecosystem was broken on my side and I did not want to fight with segfaults and resource overuse in the LSP).

If I make this proposal despite macchina having already some battery reading features is because, this library basically provide an iterator with all available batteries + update methods, and this would ease the process of reading multiple batteries.

**Comment by grtcdr** [comment]
When we fetch things, we like to call libmacchina, that's the only *backend* we wish to use. So it would be best if we wrote our own implementation, we're already borrowing quite a few things from other libraries in libmacchina and I don't like that we do that.

**Comment by naguam** [comment]
It was just a proposal and of course it would be in libmacchina,

**Comment by grtcdr** [comment]
Closing due to inactivity, you're welcome to submit a pull request, though.
